Sources/PubNub/Helpers/Crypto/CryptorModule.swift

```diff
@@ -0,0 +1,320 @@
+  
+  typealias Base64EncodedString = String
+  
+  /// Initializes `CryptorModule` with custom ``Cryptor`` objects capable of encryption and decryption
+  ///
+  /// Use this constructor if you would like to provide **custom** objects for decryption and encryption and don't want to use PubNub's built-in `Cryptors`.
+  /// Otherwise, refer to convenience static factory methods such as ``aesCbcCryptoModule(with:withRandomIV:)``
+  /// and ``legacyCryptoModule(with:withRandomIV:)`` that returns `CryptorModule` configured for you.
+  ///
+  /// - Parameters:
+  ///   - default: Primary ``Cryptor`` instance used for encryption and decryption
+  ///   - cryptors: An optional list of ``Cryptor`` instances which older messages/files were encoded
+  ///   - encoding: Default String encoding used when publishing new messages
+  public init(default cryptor: Cryptor, cryptors: [Cryptor] = [], encoding: String.Encoding = .utf8) {
+    self.defaultCryptor = cryptor
+    self.cryptors = cryptors
+    self.defaultStringEncoding = encoding
+  }
+  
+  /// Encrypts the given `Data` object
+  /// 
+  /// - Parameters:
+  ///   - data: Data to encrypt
+  /// - Returns: A success, storing encrypted `Data` if operation succeeds. Otherwise, a failure storing `PubNubError` is returned
+  public func encrypt(data: Data) -> Result<Data, PubNubError> {
+    guard !data.isEmpty else {
+      return .failure(PubNubError(
+        .encryptionError,
+        additional: ["Cannot encrypt empty Data"])
+      )
+    }
+    return defaultCryptor.encrypt(data: data).map {
+      if defaultCryptor.id == LegacyCryptor.ID {
+        return $0.data
+      } else {
+        return CryptorHeader.v1(
+          cryptorId: defaultCryptor.id,
+          data: $0.metadata
+        ).asData() + $0.data
+      }
+    }.mapError {
+      PubNubError(.encryptionError, underlying: $0)
+    }
+  }
+  
+  /// Decrypts the given `Data` object
+  ///
+  /// - Parameters:
+  ///   - data: Data to encrypt
+  /// - Returns: A success, storing decrypted `Data` if operation succeeds. Otherwise, a failure storing `PubNubError` is returned
+  public func decrypt(data: Data) -> Result<Data, PubNubError> {
+    guard !data.isEmpty else {
+      return .failure(PubNubError(
+        .decryptionError,
+        additional: ["Cannot decrypt empty Data"])
+      )
+    }
+    do {
+      let header = try CryptorHeader.from(data: data)
+      
+      guard let cryptor = cryptor(matching: header) else {
+        return .failure(PubNubError(
+          .unknownCryptorError,
+          additional: [
+            "Could not find matching Cryptor for \(header.cryptorId()) while decrypting Data. " +
+            "Ensure the corresponding instance is registered in \(String(describing: Self.self))"
+          ]
+        ))
+      }
+      return cryptor.decrypt(
+        data: EncryptedData(
+          metadata: header.metadataIfAny(),
+          data: data.subdata(in: header.length()..<data.count)
+        )
```

Comment:
If changes to `CryptorHeader` will be done (only size of cryptor-defined data stored) then here will need some changes in metadata computation (`data.subdata()`).